include/sycl/algorithm/equal.hpp

```diff
@@ -67,9 +67,9 @@ bool equal(ExecutionPolicy& exec, ForwardIt1 first1, ForwardIt1 last1,
 
   auto device = q.get_device();
 
-  auto local_size = std::min(
-      device.get_info<cl::sycl::info::device::max_work_group_size>(), size1);
-  auto global_size = exec.calculateGlobalSize(size1, local_size);
+  size_t length = size1;
+  auto ndRange = exec.calculateNdRange(size1);
```

Comment:
Please make these `const` if they're not modified. Also, `length` should be declared with `auto`.